cli/internal/env/env.go

```diff
@@ -31,21 +36,25 @@ func getEnvPairsFromKeys(envKeys []string, allEnvVars map[string]string) []strin
 }
 
 // getFrameworkEnvPairs returns a slice of all key=value pairs that match the given prefix
-func getEnvPairsFromPrefix(prefix string, allEnvVars map[string]string) []string {
+func getEnvPairsFromPrefix(prefix string, allEnvVars map[string]string, blockedEnvVars []string) []string {
 	hashableFrameworkEnvPairs := []string{}
 	for k, v := range allEnvVars {
 		if strings.HasPrefix(k, prefix) {
-			hashableFrameworkEnvPairs = append(hashableFrameworkEnvPairs, fmt.Sprintf("%v=%v", k, v))
+			for _, blocked := range blockedEnvVars {
+				if !strings.Contains(k, blocked) {
+					hashableFrameworkEnvPairs = append(hashableFrameworkEnvPairs, fmt.Sprintf("%v=%v", k, v))
+				}
+			}
 		}
 	}
 	return hashableFrameworkEnvPairs
 }
 
 // getEnvPairsFromPrefixes returns a slice containing key=value pairs for all frameworks
-func getEnvPairsFromPrefixes(prefixes []string, allEnvVars map[string]string) []string {
+func getEnvPairsFromPrefixes(prefixes []string, allEnvVars map[string]string, blockedEnvVars []string) []string {
```

Comment:
I wanted to keep this function and `getEnvPairsFromPrefix` pure and not reference the global `blockedEnvVarList` so it could continue to be generic. But `GetHashableEnvPairs` has a clear use in the name, so in my opinion it makes sense to inject this there.